### ISSUE TITLE
very interesting

### ISSUE THREAD

**Issue by robertramey** [BoostGSoC14/boost.http#7]
I believe this would be interesting to many people.  Consider adding this to the Boost Library Incubator (www.blincubator.com)  To do this, all you'd have to do is make a couple minor adjustments in directory layout and provide browsable html documentation.

Robert Ramey

**Comment by vinipsmaker** [comment]
Hi @robertramey,

I'm struggling to fill that library submission form of yours. I keep getting a "_Failed Validation - Library Submission_" error, but it won't even tell me which field is wrong.

Here is the info that I'm trying to send:

- Library Name: `Http`
- Author: `Vinícius dos Santos Oliveira`
- Short Description: `This library can be used to create applications that need to expose services through HTTP (e.g. embeddable ReST services). It's designed for multiple backends (e.g. embedded servers, FastCGI...) and has well documented type requirements and type erased abstractions, allowing compile-time-based and runtime-based polymorphism.`
- Description: `This library was envisioned to solve the HTTP problem for the C++ programming language. The proposal to solve this problem was to expose an HTTP abstraction able to make use of the HTTP power (chunking/streaming, pipelining, upgrade for supporting channels and multiplexing for supporting channels), at the same time that a complete separation of communication channels and HTTP messages is achieved.

Under this model, we refer to any abstraction fulfilling the requirements for an HTTP socket as HTTP producer. HTTP producers will feed and process HTTP messages to interested parties. The HTTP consumers are user-level components that will make use of the HTTP messages. The idea to separate communication channels and messages was proposed to handle pipelining and multiplexing elegantly, without the need to complicate the application (HTTP consumers) logic.

The ability to query certain properties from the underlying communication channel is necessary to achieve reliability under this model. It's important to define this property system this early because collaboration with the HTTP producers is key in such system. If native stream is not supported, for instance, the application would exhaust the system memory trying to stream a live video.

Besides providing a number of abstractions to isolate HTTP producers and HTTP consumers, there is also a number of abstractions on top of that to ease the job of creating applications making use of this distributed collaborative protocol.`
- Copyright Date: `2014,2015`
- First Release Data: `2014/08/18`
- Current Release Number: `0.1`
- C++ Standard: `11`
- Post Tags: Input/Output
- Library Status: `Submitted` (I don't really know what should I put here). I'm submitting to Boost Incubator first and I haven't submitted to Boost yet.
- Compilers Tested by the Author: `clang,gcc`
- Build and Linkage (what does this mean?): `CMake`
- First Order Dependencies (also confusing): `date_time,filesystem,system,unit_test_framework,context,coroutine`
- Documentation: `http://boostgsoc14.github.io/boost.http/`
- Repository: `https://github.com/BoostGSoC14/boost.http`
- Issues: `https://github.com/BoostGSoC14/boost.http/issues`
- Download (a little confusing): `https://github.com/BoostGSoC14/boost.http/archive/v0.1.tar.gz`

**Comment by robertramey** [comment]
I managed to submit your information under my own user name.  I can change the owner to you.

But I can’t find you as a user.  Are you a registered user?  If not, register and send me your username.

At one time the site gave a good error message an attempt to submit without loging in.  Perhaps I broke that functionality


Robert Ramey
Robert Ramey Software Development
(805)569-3793
ramey@rrsd.com
www.rrsd.com
